pkg/placement/ha_test.go

```diff
@@ -0,0 +1,201 @@
+package placement
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/dapr/dapr/pkg/placement/raft"
+	daprtesting "github.com/dapr/dapr/pkg/testing"
+)
+
+func TestPlacementHA(t *testing.T) {
+	// Get 3 ports
+	ports, err := daprtesting.GetFreePorts(3)
+	if err != nil {
+		log.Fatalf("failed to get 3 ports: %v", err)
+		return
+	}
+
+	// Note that ports below are unused (i.e. no service is started on those ports), they are just used as identifiers with the IP address
+	testMembers := []*raft.DaprHostMember{
+		{
+			Name:     "127.0.0.1:3031",
```

Comment:
These are also hardcoded.

Reply:
As per the comment, those are unused. We just need an IP:PORT to use as identifier for objects stored in Raft.